operator/src/main/java/oracle/kubernetes/operator/helpers/PodStepContext.java

```diff
@@ -472,6 +529,8 @@ public NextAction apply(Packet packet) {
       V1Pod currentPod = getSko().getPod().get();
       if (currentPod == null) {
         return doNext(createNewPod(getNext()), packet);
+      } else if (mustPatchPod(currentPod)) {
```

Comment:
Question on the intention here (and we might need another test)...  What if the domain resource has changed such that there would both be an additional label and an additional volume?  For that use case, I believe that the pod needs to be replaced, but I think the else clause ordering would just patch the label.

Reply:
I've now changed the order of conditionals to handle that.